azurelinuxagent/ga/cgroupapi.py

```diff
@@ -342,11 +397,78 @@ def start_extension_command(self, extension_name, command, cmd_name, timeout, sh
             with self._systemd_run_commands_lock:
                 self._systemd_run_commands.remove(process.pid)
 
-    def cleanup_legacy_cgroups(self):
+
+class SystemdCgroupApiv2(_SystemdCgroupApi):
+    """
+    Cgroup v2 interface via systemd
+    """
+    def __init__(self):
+        super(SystemdCgroupApiv2, self).__init__()
+        self._root_cgroup_path = None
+        self._controllers_enabled_at_root = []
```

Comment:
Seems like these two properties are initialized as a side effect of calling get_controller_root_paths(). That does not seems correct. For example, if somebody calls is_controller_enabled_at_root() before calling get_controller_root_paths() then self._root_cgroup_path won't be initialized.